invoice/src/main/java/org/killbill/billing/invoice/dao/DefaultInvoiceDao.java

```diff
@@ -546,17 +554,47 @@ public Pagination<InvoiceModelDao> searchInvoices(final String searchKey, final
 
         final boolean isSearchKeyCurrency = isSearchKeyCurrency(searchKey);
 
+        final SearchQuery searchQuery;
+        if (isSearchKeyCurrency) {
+            searchQuery = new SearchQuery(SqlOperator.OR);
+            searchQuery.addSearchClause("currency", SqlOperator.EQ, searchKey);
+        } else if (searchKey.startsWith(SEARCH_QUERY_MARKER)) {
+            final Matcher matcher = BALANCE_QUERY_PATTERN.matcher(searchKey);
```

Comment:
Is the use case that we have to show (return) invoices matching specific balance criteria for the given tenant - and not for a specific account?

Reply:
Yes, the use-case is to filter invoices for that tenant and balance criteria, e.g., show me all unpaid invoices.

There is a branch in the code because it doesn't support additional filtering capabilities: it's either balance filter or any other filter. Cannot do: show me all unpaid invoices in the last month unfortunately. We might be able to do it, it's just a bit hard with the balance specific query...